api/v2/routes_lens.go

```diff
@@ -28,59 +27,59 @@ func (api *API) submitIndexRequest(c *gin.Context) {
 			Fail(c, err)
 			return
 		}
+		indexType = pb.IndexReq_IPLD
 	default:
 		Fail(c, errors.New(eh.InvalidObjectTypeError))
 		return
 	}
-	// send the index request
-	resp, err := api.lens.Index(context.Background(), &request.Index{
-		Type:       forms["object_type"],
-		Identifier: forms["object_identifier"],
-		// allow for optional reindexing
-		Reindex: c.PostForm("reindex") == "yes",
+
+	resp, err := api.lens.Index(context.Background(), &pb.IndexReq{
```

Comment:
we should unify contexts so that request cancellation is handled together (aka if the HTTP request is cancelled, so is the gRPC request)

```suggestion
	resp, err := api.lens.Index(c, &pb.IndexReq{
```

Reply:
oh shit didn't know that would work. nice